### PR TITLE
100 attempts to reconnect

### PR DESCRIPTION
**Pull request by tym32167**
regarding #34
Had a dosen issues because of this. So, in my code I decided to make 100 attempts to connect, if all 100 attempts will fail - so, means that connect not possible. 
Testing this for a couple of days, me (3 instances of app) + ~5 additional users. Worked fine, at least, not crashed. 
Stackoverflow very annoying exception since it is not catchable and it destroy application each time.

### CODE REVIEW

File: BattleNET/BattlEyeClient.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
```
Comment (marceldev89) [inline BattleNET/BattlEyeClient.cs]: Remove

File: BattleNET/BattlEyeClient.cs [hunk 2]
```diff
@@ -45,6 +46,13 @@ public BattlEyeClient(BattlEyeLoginCredentials loginCredentials)
 
         public BattlEyeConnectionResult Connect()
         {
+            return ConnectInternal(100);
+        }
+
+        public BattlEyeConnectionResult ConnectInternal(int counter)
+        {
+            Debug.WriteLine($"connect counter {counter}");
```
Comment (marceldev89) [inline BattleNET/BattlEyeClient.cs]: Remove

File: BattleNET/BattlEyeClient.cs [hunk 3]
```diff
@@ -91,7 +99,9 @@ public BattlEyeConnectionResult Connect()
                 if (_disconnectionType == BattlEyeDisconnectionType.ConnectionLost)
                 {
                     Disconnect(BattlEyeDisconnectionType.ConnectionLost);
-                    Connect();
+
+                    if (counter > 0) return ConnectInternal(counter - 1);
```
Comment (marceldev89) [inline BattleNET/BattlEyeClient.cs]: return statement should be on a new line with or without braces

**Comment by tym32167** [comment]
@marceldev89 sir yes sir :) Done by new commit.

**Comment by marceldev89** [comment]
Oh, forgot 1 small thingy. The ConnectInternal method should probably be private.. sorry :)

**Comment by tym32167** [comment]
@marceldev89  no worries, my bad. I wanted it private as well. Done

**Comment by marceldev89** [comment]
Thank you. Merged. :smiley:

**Pull request merged**